src/main/java/com/myhome/services/springdatajpa/HouseMemberDocumentSDJpaService.java

```diff
@@ -0,0 +1,144 @@
+    }
+
+    @Override
+    public Optional<HouseMemberDocument> findHouseMemberDocument(String memberId) {
+        HouseMember member = houseMemberRepository.findByMemberId(memberId);
+        if(member != null) {
+            return Optional.ofNullable(member.getHouseMemberDocument());
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    @Override
+    public boolean deleteHouseMemberDocument(String memberId) {
+        boolean documentDeleted = false;
+        HouseMember member = houseMemberRepository.findByMemberId(memberId);
+        if (member != null) {
+            houseMemberDocumentRepository.delete(member.getHouseMemberDocument());
+            member.setHouseMemberDocument(null);
+            houseMemberRepository.save(member);
+            documentDeleted = true;
+        }
+        return documentDeleted;
+    }
+
+    @Override
+    public boolean updateHouseMemberDocument(MultipartFile multipartFile, String memberId) throws IOException {
+        return createHouseMemberDocument(multipartFile, memberId);
+    }
+
+    @Override
+    public boolean createHouseMemberDocument(MultipartFile multipartFile, String memberId) throws IOException {
+        boolean documentCreated = false;
+        HouseMember member = houseMemberRepository.findByMemberId(memberId);
+        if (member != null) {
+            documentCreated = tryCreateDocument(multipartFile, member);
+        }
+        return documentCreated;
+    }
+
+    private boolean tryCreateDocument(MultipartFile multipartFile, HouseMember member) throws IOException {
+
+        boolean documentCreated = false;
+        File memberDocumentFile = createMemberDocumentFile(member);
+
+        BufferedImage documentImage = getImageFromMultipartFile(multipartFile);
+        if (multipartFile.getSize() < 1024 * compressionBorderSizeKBytes) {
+            ImageIO.write(documentImage, "jpg", memberDocumentFile);
+        } else {
+            compressImageToFile(memberDocumentFile, documentImage);
+        }
+        if (memberDocumentFile.length() < 1024 * maxFileSizeKBytes) {
+            addDocumentToHouseMember(memberDocumentFile, member);
+            documentCreated = true;
+        }
+
+        memberDocumentFile.delete();
+        return documentCreated;
+    }
+
+    private HouseMember addDocumentToHouseMember(File documentFile, HouseMember member) throws IOException {
+        HouseMemberDocument houseMemberDocument = saveHouseMemberDocument(documentFile);
+        member.setHouseMemberDocument(houseMemberDocument);
+        member = houseMemberRepository.save(member);
+        return member;
+    }
+
+    private HouseMemberDocument saveHouseMemberDocument(File documentFile) throws IOException {
+        HouseMemberDocument newDocument = new HouseMemberDocument();
+        newDocument.setDocumentFilename(documentFile.getName());
+        newDocument.setDocument(Files.readAllBytes(documentFile.toPath()));
```

Comment:
Constructor can be used instead of setters since you assign all fields from `HouseMemberDocument `